### PR TITLE
External Interface: Helper functions

### PR DESCRIPTION
**Pull request by jakelang**
-Helper function to switch endianness of a given byte array.
-Simpler memory access functions

### CODE REVIEW

File: src/eei.h [hunk 1]
```diff
@@ -75,6 +75,37 @@ struct EthereumInterface : ShellExternalInterface {
     }
   }
 
+  void memRead(uint32_t offset, uint8_t *dst, size_t length)
+  {
+     uint32_t i;
+     for (i = offset; i < (offset + length); i++) {
```
Comment (axic) [inline src/eei.h]: Type declaration could be folded into the for loop, though this hear is susceptible to overflow (size requirement of offset + length might be bigger than the size of i)

File: src/eei.h [hunk 2]
```diff
@@ -75,6 +75,37 @@ struct EthereumInterface : ShellExternalInterface {
...
+
+  void memWrite(uint32_t dstoffset, uint8_t *src, size_t length)
+  {
+     uint32_t i;
+     for (i = dstoffset; i < (dstoffset + length); i++) {
+     	memory.set<uint8_t>(i, *(src + i - dstoffset));
+     }
+  }
+
+  void endianSwp(uint8_t *bytes, size_t len)
```
Comment (axic) [inline src/eei.h]: I prefer non-shortened names, such as `endianSwap`.

File: src/eei.h [hunk 3]
```diff
@@ -75,6 +75,37 @@ struct EthereumInterface : ShellExternalInterface {
...
+  {
+    int i = 0;
+    int j = len - 1;
+
```
Comment (axic) [inline src/eei.h]: Needs an assert that `len` is even.

Comment (jakelang) [reply src/eei.h]: This xor reverse algorithm should accommodate all non-zero values of len, no?

**Pull request closed without merging**